src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,44 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    public static final int MAX_SIZE = 10;
+    private K[] keyArray = (K[]) new Object[MAX_SIZE];
+    private V[] valueArray = (V[]) new Object[MAX_SIZE];
+    private int size;
+
     @Override
-    public void put(K key, V value) {
+    public void put(Object key, Object value) {
+        if (checkIfAlreadyExists(key, (V) value)) {
+            return;
+        }
+        keyArray[size] = (K) key;
+        valueArray[size] = (V) value;
+        size++;
+    }
+
+    private boolean checkIfAlreadyExists(Object key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keyArray[i] == key || key != null && key.equals(keyArray[i])) {
+                valueArray[i] = value;
+                return true;
+            }
+        }
+        return false;
```

Comment:
```suggestion
    private int getIndex(K key) {
        for (int i = 0; i < size; i++) {
            if (keyArray[i] == key || key != null && key.equals(keyArray[i])) {
                return i;
            }
            return -1;
        }
```

and we can use this method in put method and get method